src/coins.h

```diff
@@ -115,13 +115,20 @@ struct CCoinsCacheEntry
     unsigned char flags;
 
     enum Flags {
-        DIRTY = (1 << 0), // This cache entry is potentially different from the version in the parent view.
-        FRESH = (1 << 1), // The parent view does not have this entry (or it is pruned).
-        /* Note that FRESH is a performance optimization with which we can
-         * erase coins that are fully spent if we know we do not need to
-         * flush the changes to the parent cache.  It is always safe to
-         * not mark FRESH if that condition is not guaranteed.
-         */
+        //! This cache entry is potentially different from the version in the parent cache.
+        //! Failing to mark a coin as DIRTY when it is potentially different from the
+        //! parent view will cause a consensus failure, since the coin's state won't get
+        //! written to the parent when the cache is flushed.
+        //! Note that all empty coins in the cache MUST be DIRTY.
```

Comment:
What is an "empty" coin?

Reply:
I think this references `emptyCoin` (`coins.h:132`) which calls an empty contractor of `Coin` (`coins.h:53`).

Reply:
> I think this references `emptyCoin` (`coins.h:132`) which calls an empty contractor of `Coin` (`coins.h:53`).

I think this is suppose to say `coinEmpty` (`coins.cpp:132`).

Reply:
You're right that this is unclear. Empty here means that the coin has been spent (see Coin.Clear() above which nulls out a coin when it is spent). I've changed this to "Note that a spent coin MUST be DIRTY", which I think is much clearer.